### ISSUE TITLE
Instance Segmentation performance?

### ISSUE THREAD

**Issue by happywu** [facebookresearch/detr#149]
Hi, 
I use the following command to train instance segmentation
```
python -m torch.distributed.launch --nproc_per_node=4  --use_en main.py --epochs 25 --lr_drop 15 --coco_path $COCO/data --frozen_weights pretrained/detr-r50-e632da11.pth  --batch_size 2 --masks --no_aux_loss
```

The trained model only has a Seg AP of 24.7, is the normal?

**Comment by alcinos** [comment]
Hi @happywu 
This sounds like a low performance, I'd expect at least 30 AP without too much trouble. Are you training on only one node? If so your effective batch-size (8) is quite small, you might need to adjust a bit the learning rates accordingly.

Best of luck.

**Comment by happywu** [comment]
yes, 4 gpus are used. I'll try adjust learning rates / # gpus.

Thanks.

**Comment by happywu** [comment]
Hi @alcinos 
Changing to 8 gpus gives Segm Ap of 26.1
Is this because the framework is not suitable for instance seg or anything else do I have to tune?

Thanks.

**Comment by alcinos** [comment]
Hi again,
Just testing this with batch-size = 2 on 32 cards gives 27.5 AP in only 5 epochs (still going up, will report what I get after 25 epochs).
I still think this is a learning rate scaling issue.
Best of luck

**Comment by alcinos** [comment]
Final performance:
IoU metric: segm
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.332
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.565
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.339
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.135
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.368
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.531
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.280
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.433
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.462
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.234
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.515
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.666

**Comment by happywu** [comment]
Thanks! The performance looks pretty good. 
So you run 25 epochs, with head learning rate of 1e-4, and freeze the backbone, using 32 gpus, right?
Do you have recommended learning rate for 8 gpus? Learning rescaling seems not work with Adam. 

Thanks!
Haiping

**Comment by happywu** [comment]
Hi @alcinos ,

I managed to use 8 gpus, with learning rate of 5e-5 to get 33.5.
Maybe could update readme.

IoU metric: segm
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.335
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.568
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.343
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.139
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.370
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.535
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.283
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.436
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.465
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.234
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.518
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.671

**Comment by anirudh-chakravarthy** [comment]
Hi @alcinos 

Is it possible to provide the pre-trained instance segmentation weights? I'm trying to perform instance segmentation, and I'm not able to find pre-trained weights to initialize the mask head.

**Comment by trqminh** [comment]
Hi @alcinos ,
I wonder what is the instance segmentation head is used here? Is it a transformer-based or conv-based?
Thank you for your consideration.
